### ISSUE TITLE
Metal Stoker (Japan) crash

### ISSUE THREAD

**Issue by sarahwhitman** [libretro/beetle-pce-fast-libretro#186]
Hello all! Reporting a bug that affects the Japanese PC-Engine release "Metal Stoker" - the game consistently crashes after 10-15 seconds of gameplay. This issue **does not** affect beetle-pce-libretro core, only the 'fast' variant. I have checked upstream for a mednafen bug report on this issue but didn't find anything. The bug affects all builds of beetle-pce-fast-libretro that I have tested.

Tested versions:
Retroarch 1.9.0, 1.9.1
beetle-pce-fast-libretro 0.9.38.7 b29972b  -  and other nightly builds dating back to December 2020

My machine:
Dell Precision 5520
Windows 10 x64 1809 LTSC, build 17763.1879
Intel HD 630 graphics: driver 27.20.100.9171, date 1/8/2021

Core options:
pce_adpcmvolume = "100"
pce_cddavolume = "100"
pce_cdpsgvolume = "100"
pce_cdspeed = "1"
pce_disable_softreset = "disabled"
pce_fast_cdbios = "System Card 3"
pce_fast_cdimagecache = "disabled"
pce_fast_frameskip = "disabled"
pce_fast_frameskip_threshold = "33"
pce_fast_palette = "Composite"
pce_hoverscan = "352"
pce_initial_scanline = "0"
pce_last_scanline = "242"
pce_mouse_sensitivity = "1.25"
pce_nospritelimit = "disabled"
pce_ocmultiplier = "1"
pce_sound_channel_0_volume = "100"
pce_sound_channel_1_volume = "100"
pce_sound_channel_2_volume = "100"
pce_sound_channel_3_volume = "100"
pce_sound_channel_4_volume = "100"
pce_sound_channel_5_volume = "100"
pce_turbo_delay = "3"
pce_turbo_toggle_hotkey = "disabled"
pce_turbo_toggling = "disabled"

**Comment by negativeExponent** [comment]
crash to desktop? or game looks like this (or any other random images on-screen):

[img]https://i.imgur.com/jFgLT7M.png[/img]

**Comment by sarahwhitman** [comment]
It is more of a lockup than a crash - the core is still running with the audio stuck in a loop. Retroarch is still responsive and I can close / restart the content. On prior builds I've seen graphics corruption like what you linked, usually just before it locks up. On the current build, I'm not able to reproduce the graphics corruption at the moment.

**Comment by negativeExponent** [comment]
well then, i found a separate bug.. but i cannot replicate your issue. the game just has to run 10-15 right? i've played more than that and no crash of any form. what platform are you playing on?

**Comment by sarahwhitman** [comment]
Yes the lockup occurs after no more than 15 seconds gameplay. I just tested again and finally was able capture some examples of graphics corruption. In one instance the game went black and white just before freezing, in the second and third screenshots the screen slowly started to garble before freezing.

![Metal Stoker (Japan)-210428-154438](https://user-images.githubusercontent.com/65987867/116481964-155dc080-a839-11eb-8939-58bb0ba10f6e.png)
![Metal Stoker (Japan)-210428-154516](https://user-images.githubusercontent.com/65987867/116481967-17c01a80-a839-11eb-9b21-571700d73404.png)
![Metal Stoker (Japan)-210428-154522](https://user-images.githubusercontent.com/65987867/116481971-1989de00-a839-11eb-91c1-b44124bd11c7.png)

All of these are on a Windows 10 x64 platform - its a pretty standard system running only Intel integrated graphics. I'm happy to collect logs/config files/etc. if you let me know what you need!
